project-base/storefront/hooks/cart/useRemoveFromCart.ts

```diff
@@ -1,26 +1,36 @@
-import { CartFragmentApi, CartItemFragmentApi, useRemoveFromCartMutationApi } from 'graphql/generated';
+import { CartItemFragmentApi, useCartQueryApi, useRemoveFromCartMutationApi } from 'graphql/generated';
 import { onGtmRemoveFromCartEventHandler } from 'gtm/helpers/eventHandlers';
 import { GtmProductListNameType } from 'gtm/types/enums';
 import { mapPriceForCalculations } from 'helpers/mappers/price';
+import { useIsUserLoggedIn } from 'hooks/auth/useIsUserLoggedIn';
 import { useDomainConfig } from 'hooks/useDomainConfig';
 import { usePersistStore } from 'store/usePersistStore';
 
-export type RemoveFromCartHandler = (
-    cartItem: CartItemFragmentApi,
-    listIndex: number,
-) => Promise<CartFragmentApi | undefined | null>;
+export type RemoveFromCartHandler = (cartItem: CartItemFragmentApi, listIndex: number) => Promise<void>;
 
 export const useRemoveFromCart = (gtmProductListName: GtmProductListNameType): [RemoveFromCartHandler, boolean] => {
     const [{ fetching }, removeItemFromCart] = useRemoveFromCartMutationApi();
     const { url, currencyCode } = useDomainConfig();
     const cartUuid = usePersistStore((store) => store.cartUuid);
+    const isUserLoggedIn = useIsUserLoggedIn();
+
+    const [, refetchCart] = useCartQueryApi({
+        variables: { cartUuid },
+        pause: !cartUuid && !isUserLoggedIn,
+        requestPolicy: 'network-only',
+    });
```

Comment:
It is not calling CartQuery twice, because it's using cached data for second call. We have two calls of useCartQueryApi which is nothing what we wouldn't already used in our app. But I forgot I've already exported `refetchCart` from `useCurrentCart`. So I am using this exported function and this hook is removed.

Reply:
It is calling cart twice, still. See the video


Uploading Screen Recording 2023-11-03 at 10.11.41.mov…

Reply:
Solved now. Thanks for your help!